optimizely/config/polling_manager.go

```diff
@@ -29,31 +29,67 @@ import (
 	"github.com/optimizely/go-sdk/optimizely/utils"
 )
 
-const defaultPollingInterval = 5 * time.Minute // default to 5 minutes for polling
+// DefaultPollingInterval sets default interval for polling manager
+const DefaultPollingInterval = 5 * time.Minute // default to 5 minutes for polling
 
 // DatafileURLTemplate is used to construct the endpoint for retrieving the datafile from the CDN
 const DatafileURLTemplate = "https://cdn.optimizely.com/datafiles/%s.json"
 
 var cmLogger = logging.GetLogger("PollingConfigManager")
 
-// PollingProjectConfigManagerOptions used to create an instance with custom configuration
-type PollingProjectConfigManagerOptions struct {
-	Datafile           []byte
-	PollingInterval    time.Duration
-	Requester          utils.Requester
-	NotificationCenter notification.Center
-}
-
 // PollingProjectConfigManager maintains a dynamic copy of the project config
 type PollingProjectConfigManager struct {
 	requester          utils.Requester
 	pollingInterval    time.Duration
-	projectConfig      optimizely.ProjectConfig
-	configLock         sync.RWMutex
-	err                error
 	notificationCenter notification.Center
+	initDatafile       []byte
+
+	configLock    sync.RWMutex
+	err           error
+	projectConfig optimizely.ProjectConfig
+	exeCtx        utils.ExecutionCtx // context used for execution control
+}
+
+// OptionFunc is a type to a proper func
+type OptionFunc func(*PollingProjectConfigManager)
+
+// SetDefaultRequester is an optional function, sets default requester based on a key.
+func SetDefaultRequester(sdkKey string) OptionFunc {
```

Comment:
It's inconsistent that the optionfunc for the polling manager starts with "Set" while the ones in the optly client don't. Let's get to keep them consistent.

Reply:
done